mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -3125,6 +3153,41 @@ void MaskedOp::build(OpBuilder &builder, OperationState &state,
   MaskedOp::build(builder, state, clauses.filteredThreadId);
 }
 
+//===----------------------------------------------------------------------===//
+// Spec 5.2: Scan construct (5.6)
+//===----------------------------------------------------------------------===//
+
+void ScanOp::build(OpBuilder &builder, OperationState &state,
+                   const ScanOperands &clauses) {
+  ScanOp::build(builder, state, clauses.inclusiveVars, clauses.exclusiveVars);
+}
+
+LogicalResult ScanOp::verify() {
+  if (hasExclusiveVars() && hasInclusiveVars()) {
+    return emitError(
+        "Exactly one of EXCLUSIVE or INCLUSIVE clause is expected");
+  }
+  if (mlir::omp::WsloopOp parentWsLoopOp =
```

Comment:
Nit: We're already in the `mlir::omp` namespace (and there's a `using mlir::omp` at the top of the file). There are a few more instances in this function.
```suggestion
  if (WsloopOp parentWsLoopOp =
```